### ISSUE TITLE
NVDA not reading in specified "lang" when inside an element with an aria role

### ISSUE THREAD

**Issue by kiahnjl** [nvaccess/nvda#9181]
### Steps to reproduce:
Use this HTML and run NVDA, then tab to the `p` element:
```
<html lang="en">
  <body>
    <div role="application">
      <p lang="fr" tabindex="0">Ce paragraphe est défini en français.</p> 
    </div>
  </body>
</html>
```

### Actual behavior:
The paragraph is read out in English.

### Expected behavior:
The paragraph should be read out in French.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2018.4.1

#### Windows version:
10.0.17134

#### Does the issue still occur after restarting your PC?
Yes

**Comment by florabazie** [comment]
Hi Kiahnjl. I have encountered the same issue recently in one project. 
To carefully check the failure of NVDA for language of parts, I created a simple plain text html page (see plain text attached to be saved as html format)
I tested the page with JAWS and NVDA. JAWS result was positive for French. NVDA was a failure.
I even set a French paragraph with English lang attribute to see and compared with other French paragraph read by NVDA. Not surprised to see that it was rendered the same.

Can you remove the role="application" from your example and run NVDA? Does this make a difference?

Thank you for sharing this issue

[language_of_parts.txt](https://github.com/nvaccess/nvda/files/3334930/language_of_parts.txt)

**Comment by Adriani90** [comment]
The language is recognized properly in browse mode, but not in focus mode. Tested with NVDA 2020.1 on Chrome 81 and Edge Chromium 81.

**Comment by Adriani90** [comment]
@jcsteh any idea why in Firefox NVDA does not read the content of the paragraph in focus mode despite its tabindex="0"? Is this intentional? In chromium the tabindex is respected in this case.

I applied role="link" on the paragraph to test the same conditions like in Chromium, and it seems NVDA would still fail to recognize the language in focus mode also in Firefox.
data:text/html,<html lang="en"><body><div role="application"><p role="link" lang="fr" tabindex="0">Ce paragraphe est d%C3%A9fini en fran%C3%A7ais.</p></div></body></html>

**Comment by jcsteh** [comment]
Per [the ARIA spec](https://w3c.github.io/aria/#paragraph), a paragraph cannot have a name (Name from: prohibited). There is some ongoing debate about whether browsers should ignore this for focusable elements, but that hasn't been ratified in the spec. Chromium is choosing to deviate from the spec here. There is some sense in that, but also some potential problems, and it's not a change I'm willing to make in Firefox unless it gets specified.

As for the reported issue here, role="application" means that browse mode isn't used, and NVDA doesn't currently support language switching outside of text in browse mode. That makes this a duplicate of #3188.
